From 326e866790a663cadaa323d09ed7ead36a0e2256 Mon Sep 17 00:00:00 2001
From: Vitaly Kuznetsov <vkuznets@redhat.com>
Date: Thu, 29 May 2014 16:49:36 -0400
Subject: [virt] Revert: xen/spinlock: Disable IRQ spinlock (PV) allocation on PVHVM

Message-id: <1401382177-12769-6-git-send-email-vkuznets@redhat.com>
Patchwork-id: 81163
O-Subject: [RHEL7.1 PATCH 5/6] Revert "xen/spinlock: Disable IRQ spinlock (PV) allocation on PVHVM"
Bugzilla: 1006977
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Andrew Jones <drjones@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1006977
Upstream commit: fb78e58c275599a961557bca201dd897565833ca
Brew build: https://brewweb.devel.redhat.com/taskinfo?taskID=7482680

Author: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Date:   Mon Aug 26 14:28:50 2013 -0400

This reverts commit 70dd4998cb85f0ecd6ac892cc7232abefa432efb.

Now that the bugs have been resolved we can re-enable the
PV ticketlock implementation under PVHVM Xen guests.

Signed-off-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Reviewed-by: David Vrabel <david.vrabel@citrix.com>
Signed-off-by: Vitaly Kuznetsov <vkuznets@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/xen/spinlock.c b/arch/x86/xen/spinlock.c
index e1bff87..b3c1ee8 100644
--- a/arch/x86/xen/spinlock.c
+++ b/arch/x86/xen/spinlock.c
@@ -230,13 +230,6 @@ void xen_init_lock_cpu(int cpu)
  WARN(per_cpu(lock_kicker_irq, cpu) >= 0, "spinlock on CPU%d exists on IRQ%d!\n",
       cpu, per_cpu(lock_kicker_irq, cpu));
 
- /*
-  * See git commit f10cd522c5fbfec9ae3cc01967868c9c2401ed23
-  * (xen: disable PV spinlocks on HVM)
-  */
- if (xen_hvm_domain())
-  return;
-
  name = kasprintf(GFP_KERNEL, "spinlock%d", cpu);
  irq = bind_ipi_to_irqhandler(XEN_SPIN_UNLOCK_VECTOR,
          cpu,
@@ -256,13 +249,6 @@ void xen_init_lock_cpu(int cpu)
 
 void xen_uninit_lock_cpu(int cpu)
 {
- /*
-  * See git commit f10cd522c5fbfec9ae3cc01967868c9c2401ed23
-  * (xen: disable PV spinlocks on HVM)
-  */
- if (xen_hvm_domain())
-  return;
-
  if (!xen_pvspin)
   return;
 
@@ -275,12 +261,6 @@ void xen_uninit_lock_cpu(int cpu)
 
 void __init xen_init_spinlocks(void)
 {
- /*
-  * See git commit f10cd522c5fbfec9ae3cc01967868c9c2401ed23
-  * (xen: disable PV spinlocks on HVM)
-  */
- if (xen_hvm_domain())
-  return;
 
  if (!xen_pvspin) {
   printk(KERN_DEBUG "xen: PV spinlocks disabled\n");
-- 
1.7.1